coap-gateway/service/observation/resourcesObserver.go

```diff
@@ -47,12 +47,12 @@ func MakeResourcesObserverCallbacks(onObserveResource OnObserveResource, onGetRe
 //     the content of the resource and the response is handled by the onGetResourceContent
 //     callback.
 type resourcesObserver struct {
-	lock      sync.Mutex
-	deviceID  string
-	resources observedResources
-	coapConn  ClientConn
 	callbacks ResourcesObserverCallbacks
+	coapConn  ClientConn
 	logger    log.Logger
+	deviceID  string
+	resources observedResources
```

Comment:
`// guarded by lock`

I think only this one member is protected, the others are set in the constructor and never changed.

Reply:
moved to private struct